src/level_four/core/controller/send_transaction_controller_solana.c

```diff
@@ -175,7 +175,31 @@ void send_transaction_controller_solana() {
     } break;
 
     case SEND_TXN_TAP_CARD_SEND_CMD_SOLANA: {
-      flow_level.level_three = SEND_TXN_READ_DEVICE_SHARE_SOLANA;
+      flow_level.level_three = SEND_TXN_UPDATE_BLOCKHASH_SOLANA;
+      uint8_t arr[1]         = {1};
+      transmit_data_to_app(SEND_TXN_PRE_SIGNING_DATA, arr, sizeof(arr));
+    } break;
+
+    case SEND_TXN_UPDATE_BLOCKHASH_SOLANA: {
+      uint8_t *blockhash_received                              = NULL;
+      uint16_t blockhash_msg_size                              = 0;
+      uint8_t solana_latest_blockhash[SOLANA_BLOCKHASH_LENGTH] = {0};
+
+      if (get_usb_msg_by_cmd_type(SEND_TXN_PRE_SIGNING_DATA, &blockhash_received, &blockhash_msg_size)) {
+        if (blockhash_msg_size != SOLANA_BLOCKHASH_LENGTH) {
+          comm_reject_invalid_cmd();
+          reset_flow_level();
+        }
+        memcpy(solana_latest_blockhash, blockhash_received, SOLANA_BLOCKHASH_LENGTH);
+        clear_message_received_data();
+
+        int status = solana_update_blockhash_in_byte_array(solana_unsigned_txn_byte_array, solana_latest_blockhash);
+        if (status != SEC_OK) {
```

Comment:
SEC should be replaced with SOL overall for solana error codes.
Either add change in this MR or raise another